{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 362,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "from sklearn.preprocessing import StandardScaler\n",
    "\n",
    "%matplotlib inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 363,
   "metadata": {},
   "outputs": [],
   "source": [
    "# read the sequences file\n",
    "arquivo = 'hikoyu_fasta.txt'\n",
    "\n",
    "file_content = open(arquivo, \"r\")\n",
    "\"\"\"file_content.close() \"\"\"\n",
    "\n",
    "all_lines = file_content.readlines();\n",
    "\n",
    "sequences = {}\n",
    "sequences_info = {}\n",
    "description = ''\n",
    "\n",
    "for line in all_lines:\n",
    "    \n",
    "    sequence = []\n",
    "    \n",
    "    if \">\" in line:\n",
    "        description = line\n",
    "        sequence = []\n",
    "        sequences[description] = sequence\n",
    "    else:\n",
    "        sequences[description].append(line)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 364,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"sequences['>MT126808.1 Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/SP02/human/2020/BRA, complete genome\\n']\""
      ]
     },
     "execution_count": 364,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'''sequences['>MT126808.1 Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/SP02/human/2020/BRA, complete genome\\n']'''"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 370,
   "metadata": {},
   "outputs": [],
   "source": [
    "#count A,T,C,G,AA,AT,AC,AG,TT,TA,TC,TG,CC,CA,CT,CG,GG,GA,GT,GC for each sequence\n",
    "\n",
    "for description in sequences:\n",
    "    \n",
    "    sequences_info[description] = {}\n",
    "    \n",
    "    sequences_info[description]['A'] = 0\n",
    "    sequences_info[description]['T'] = 0\n",
    "    sequences_info[description]['C'] = 0\n",
    "    sequences_info[description]['G'] = 0\n",
    "    \n",
    "    sequences_info[description]['AA'] = 0\n",
    "    sequences_info[description]['AT'] = 0\n",
    "    sequences_info[description]['AC'] = 0\n",
    "    sequences_info[description]['AG'] = 0\n",
    "    \n",
    "    sequences_info[description]['TA'] = 0\n",
    "    sequences_info[description]['TT'] = 0\n",
    "    sequences_info[description]['TC'] = 0\n",
    "    sequences_info[description]['TG'] = 0\n",
    "    \n",
    "    sequences_info[description]['CA'] = 0\n",
    "    sequences_info[description]['CT'] = 0\n",
    "    sequences_info[description]['CC'] = 0\n",
    "    sequences_info[description]['CG'] = 0\n",
    "    \n",
    "    sequences_info[description]['GA'] = 0\n",
    "    sequences_info[description]['GT'] = 0\n",
    "    sequences_info[description]['GC'] = 0\n",
    "    sequences_info[description]['GG'] = 0\n",
    "    \n",
    "    sequences_info[description]['SIZE'] = 0\n",
    "    \n",
    "    #key = '>MT126808.1 Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/SP02/human/2020/BRA, complete genome\\n'\n",
    "\n",
    "    for line in sequences[description]:\n",
    "        \n",
    "        sequences_info[description]['SIZE'] += len(line)\n",
    "        sequences_info[description]['A'] += line.count('A')\n",
    "        sequences_info[description]['T'] += line.count('T')\n",
    "        sequences_info[description]['C'] += line.count('C')\n",
    "        sequences_info[description]['G'] += line.count('G')\n",
    "        \n",
    "        sequences_info[description]['AA'] += line.count('AA')\n",
    "        sequences_info[description]['AT'] += line.count('AT')\n",
    "        sequences_info[description]['AC'] += line.count('AC')\n",
    "        sequences_info[description]['AG'] += line.count('AG')\n",
    "    \n",
    "        sequences_info[description]['TA'] += line.count('AT')\n",
    "        sequences_info[description]['TT'] += line.count('TT')\n",
    "        sequences_info[description]['TC'] += line.count('TC')\n",
    "        sequences_info[description]['TG'] += line.count('TG')\n",
    "    \n",
    "        sequences_info[description]['CA'] += line.count('CA')\n",
    "        sequences_info[description]['CT'] += line.count('CT')\n",
    "        sequences_info[description]['CC'] += line.count('CC')\n",
    "        sequences_info[description]['CG'] += line.count('CG')\n",
    "    \n",
    "        sequences_info[description]['GA'] += line.count('GA')\n",
    "        sequences_info[description]['GT'] += line.count('GT')\n",
    "        sequences_info[description]['GC'] += line.count('GC')\n",
    "        sequences_info[description]['GG'] += line.count('GG')   \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 371,
   "metadata": {},
   "outputs": [],
   "source": [
    "#calc nucleotideo bias\n",
    "\n",
    "for description in sequences:\n",
    "    \n",
    "    sequences_info[description]['pAA'] = 0\n",
    "    sequences_info[description]['pAT'] = 0\n",
    "    sequences_info[description]['pAC'] = 0\n",
    "    sequences_info[description]['pAG'] = 0\n",
    "    \n",
    "    sequences_info[description]['pTA'] = 0\n",
    "    sequences_info[description]['pTT'] = 0\n",
    "    sequences_info[description]['pTC'] = 0\n",
    "    sequences_info[description]['pTG'] = 0\n",
    "    \n",
    "    sequences_info[description]['pCA'] = 0\n",
    "    sequences_info[description]['pCT'] = 0\n",
    "    sequences_info[description]['pCC'] = 0\n",
    "    sequences_info[description]['pCG'] = 0\n",
    "    \n",
    "    sequences_info[description]['pGA'] = 0\n",
    "    sequences_info[description]['pGT'] = 0\n",
    "    sequences_info[description]['pGC'] = 0\n",
    "    sequences_info[description]['pGG'] = 0\n",
    "         \n",
    "    fA = sequences_info[description]['A'] / sequences_info[description]['SIZE'] \n",
    "    fT = sequences_info[description]['T'] / sequences_info[description]['SIZE']\n",
    "    fC = sequences_info[description]['C'] / sequences_info[description]['SIZE']\n",
    "    fG = sequences_info[description]['G'] / sequences_info[description]['SIZE'] \n",
    "        \n",
    "    fAA = sequences_info[description]['AA'] / sequences_info[description]['SIZE']       \n",
    "    fAT = sequences_info[description]['AT'] / sequences_info[description]['SIZE']       \n",
    "    fAC = sequences_info[description]['AC'] / sequences_info[description]['SIZE']       \n",
    "    fAG = sequences_info[description]['AG'] / sequences_info[description]['SIZE']    \n",
    "      \n",
    "    fTA = sequences_info[description]['TA'] / sequences_info[description]['SIZE']       \n",
    "    fTT = sequences_info[description]['TT'] / sequences_info[description]['SIZE']       \n",
    "    fTC = sequences_info[description]['TC'] / sequences_info[description]['SIZE']       \n",
    "    fTG = sequences_info[description]['TG'] / sequences_info[description]['SIZE']       \n",
    "\n",
    "    fCA = sequences_info[description]['CA'] / sequences_info[description]['SIZE']       \n",
    "    fCT = sequences_info[description]['CT'] / sequences_info[description]['SIZE']       \n",
    "    fCC = sequences_info[description]['CC'] / sequences_info[description]['SIZE']       \n",
    "    fCG = sequences_info[description]['CG'] / sequences_info[description]['SIZE']       \n",
    "\n",
    "    fGA = sequences_info[description]['GA'] / sequences_info[description]['SIZE']       \n",
    "    fGT = sequences_info[description]['GT'] / sequences_info[description]['SIZE']       \n",
    "    fGC = sequences_info[description]['GC'] / sequences_info[description]['SIZE']       \n",
    "    fGG = sequences_info[description]['GG'] / sequences_info[description]['SIZE']\n",
    "        \n",
    "    sequences_info[description]['pA'] = fA\n",
    "    sequences_info[description]['pT'] = fT\n",
    "    sequences_info[description]['pC'] = fC\n",
    "    sequences_info[description]['pG'] = fG\n",
    "       \n",
    "    sequences_info[description]['pAA'] = fAA/fA*fA\n",
    "    sequences_info[description]['pAT'] = fAT/fA*fT\n",
    "    sequences_info[description]['pAC'] = fAC/fA*fC\n",
    "    sequences_info[description]['pAG'] = fAC/fA*fG\n",
    "\n",
    "    sequences_info[description]['pTA'] = fTA/fT*fA\n",
    "    sequences_info[description]['pTT'] = fTT/fT*fT\n",
    "    sequences_info[description]['pTC'] = fTC/fT*fC\n",
    "    sequences_info[description]['pTG'] = fTC/fT*fG\n",
    "      \n",
    "    sequences_info[description]['pCA'] = fCA/fC*fA\n",
    "    sequences_info[description]['pCT'] = fCT/fC*fT\n",
    "    sequences_info[description]['pCC'] = fCC/fC*fC\n",
    "    sequences_info[description]['pCG'] = fCC/fC*fG\n",
    "\n",
    "    sequences_info[description]['pGA'] = fGA/fG*fA\n",
    "    sequences_info[description]['pGT'] = fGT/fG*fT\n",
    "    sequences_info[description]['pGC'] = fGC/fG*fC\n",
    "    sequences_info[description]['pGG'] = fGC/fG*fG\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 372,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'A': 8927,\n",
       " 'T': 9596,\n",
       " 'C': 5492,\n",
       " 'G': 5861,\n",
       " 'AA': 2126,\n",
       " 'AT': 2275,\n",
       " 'AC': 1987,\n",
       " 'AG': 1705,\n",
       " 'TA': 2275,\n",
       " 'TT': 2424,\n",
       " 'TC': 1392,\n",
       " 'TG': 2543,\n",
       " 'CA': 2039,\n",
       " 'CT': 2031,\n",
       " 'CC': 773,\n",
       " 'CG': 430,\n",
       " 'GA': 1591,\n",
       " 'GT': 1957,\n",
       " 'GC': 1151,\n",
       " 'GG': 962,\n",
       " 'SIZE': 30374,\n",
       " 'pAA': 0.06999407387897544,\n",
       " 'pAT': 0.08051264918852753,\n",
       " 'pAC': 0.04024582850845922,\n",
       " 'pAG': 0.042949890911886286,\n",
       " 'pTA': 0.0696778446046512,\n",
       " 'pTT': 0.07980509646408113,\n",
       " 'pTC': 0.0262287465149284,\n",
       " 'pTG': 0.02799102027021037,\n",
       " 'pCA': 0.10911645381723578,\n",
       " 'pCT': 0.11683356951653605,\n",
       " 'pCC': 0.02544939751102917,\n",
       " 'pCG': 0.027159307868197737,\n",
       " 'pGA': 0.07978146455527685,\n",
       " 'pGT': 0.10548904361906612,\n",
       " 'pGC': 0.035508484922544464,\n",
       " 'pGG': 0.037894251662606175,\n",
       " 'pA': 0.2939026799236189,\n",
       " 'pT': 0.3159280963982353,\n",
       " 'pC': 0.1808125370382564,\n",
       " 'pG': 0.19296108513860538}"
      ]
     },
     "execution_count": 372,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sequences_info['>MT126808.1 Severe acute respiratory syndrome coronavirus 2 isolate SARS-CoV-2/SP02/human/2020/BRA, complete genome\\n']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 373,
   "metadata": {},
   "outputs": [],
   "source": [
    "#creating a dataframe\n",
    "df_original = pd.DataFrame.from_dict(sequences_info, orient='index', columns=['pA', 'pT', 'pC', 'pG',\n",
    "                                                                     'pAA','pAT', 'pAC', 'pAG',\n",
    "                                                                     'pTA','pTA', 'pTC', 'pTG',\n",
    "                                                                     'pCA','pCT', 'pCC', 'pCG',\n",
    "                                                                     'pGA','pGT', 'pGC', 'pGG'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pA</th>\n",
       "      <th>pT</th>\n",
       "      <th>pC</th>\n",
       "      <th>pG</th>\n",
       "      <th>pAA</th>\n",
       "      <th>pAT</th>\n",
       "      <th>pAC</th>\n",
       "      <th>pAG</th>\n",
       "      <th>pTA</th>\n",
       "      <th>pTA</th>\n",
       "      <th>pTC</th>\n",
       "      <th>pTG</th>\n",
       "      <th>pCA</th>\n",
       "      <th>pCT</th>\n",
       "      <th>pCC</th>\n",
       "      <th>pCG</th>\n",
       "      <th>pGA</th>\n",
       "      <th>pGT</th>\n",
       "      <th>pGC</th>\n",
       "      <th>pGG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>&gt;LC521925.1 Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/AI/I-004/2020 RNA, complete genome\\n</th>\n",
       "      <td>0.009758</td>\n",
       "      <td>0.007422</td>\n",
       "      <td>0.012507</td>\n",
       "      <td>0.000275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000926</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>0.001501</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;LC522972.1 Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/KY/V-029/2020 RNA, complete genome\\n</th>\n",
       "      <td>0.009758</td>\n",
       "      <td>0.007422</td>\n",
       "      <td>0.012507</td>\n",
       "      <td>0.000275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000926</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>0.001501</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;LC522973.1 Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-012/2020 RNA, complete genome\\n</th>\n",
       "      <td>0.009758</td>\n",
       "      <td>0.007422</td>\n",
       "      <td>0.012507</td>\n",
       "      <td>0.000275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000926</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>0.001501</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;LC522974.1 Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-501/2020 RNA, complete genome\\n</th>\n",
       "      <td>0.009758</td>\n",
       "      <td>0.007422</td>\n",
       "      <td>0.012507</td>\n",
       "      <td>0.000275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000926</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>0.001501</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;LC522975.1 Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-521/2020 RNA, complete genome\\n</th>\n",
       "      <td>0.009758</td>\n",
       "      <td>0.007422</td>\n",
       "      <td>0.012507</td>\n",
       "      <td>0.000275</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.000926</td>\n",
       "      <td>0.00002</td>\n",
       "      <td>0.001501</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                          pA        pT  \\\n",
       ">LC521925.1 Severe acute respiratory syndrome c...  0.009758  0.007422   \n",
       ">LC522972.1 Severe acute respiratory syndrome c...  0.009758  0.007422   \n",
       ">LC522973.1 Severe acute respiratory syndrome c...  0.009758  0.007422   \n",
       ">LC522974.1 Severe acute respiratory syndrome c...  0.009758  0.007422   \n",
       ">LC522975.1 Severe acute respiratory syndrome c...  0.009758  0.007422   \n",
       "\n",
       "                                                          pC        pG  pAA  \\\n",
       ">LC521925.1 Severe acute respiratory syndrome c...  0.012507  0.000275  0.0   \n",
       ">LC522972.1 Severe acute respiratory syndrome c...  0.012507  0.000275  0.0   \n",
       ">LC522973.1 Severe acute respiratory syndrome c...  0.012507  0.000275  0.0   \n",
       ">LC522974.1 Severe acute respiratory syndrome c...  0.012507  0.000275  0.0   \n",
       ">LC522975.1 Severe acute respiratory syndrome c...  0.012507  0.000275  0.0   \n",
       "\n",
       "                                                    pAT  pAC  pAG  pTA  pTA  \\\n",
       ">LC521925.1 Severe acute respiratory syndrome c...  0.0  0.0  0.0  0.0  0.0   \n",
       ">LC522972.1 Severe acute respiratory syndrome c...  0.0  0.0  0.0  0.0  0.0   \n",
       ">LC522973.1 Severe acute respiratory syndrome c...  0.0  0.0  0.0  0.0  0.0   \n",
       ">LC522974.1 Severe acute respiratory syndrome c...  0.0  0.0  0.0  0.0  0.0   \n",
       ">LC522975.1 Severe acute respiratory syndrome c...  0.0  0.0  0.0  0.0  0.0   \n",
       "\n",
       "                                                         pTC      pTG  \\\n",
       ">LC521925.1 Severe acute respiratory syndrome c...  0.000926  0.00002   \n",
       ">LC522972.1 Severe acute respiratory syndrome c...  0.000926  0.00002   \n",
       ">LC522973.1 Severe acute respiratory syndrome c...  0.000926  0.00002   \n",
       ">LC522974.1 Severe acute respiratory syndrome c...  0.000926  0.00002   \n",
       ">LC522975.1 Severe acute respiratory syndrome c...  0.000926  0.00002   \n",
       "\n",
       "                                                         pCA  pCT  pCC  pCG  \\\n",
       ">LC521925.1 Severe acute respiratory syndrome c...  0.001501  0.0  0.0  0.0   \n",
       ">LC522972.1 Severe acute respiratory syndrome c...  0.001501  0.0  0.0  0.0   \n",
       ">LC522973.1 Severe acute respiratory syndrome c...  0.001501  0.0  0.0  0.0   \n",
       ">LC522974.1 Severe acute respiratory syndrome c...  0.001501  0.0  0.0  0.0   \n",
       ">LC522975.1 Severe acute respiratory syndrome c...  0.001501  0.0  0.0  0.0   \n",
       "\n",
       "                                                    pGA  pGT  pGC  pGG  \n",
       ">LC521925.1 Severe acute respiratory syndrome c...    0    0    0    0  \n",
       ">LC522972.1 Severe acute respiratory syndrome c...    0    0    0    0  \n",
       ">LC522973.1 Severe acute respiratory syndrome c...    0    0    0    0  \n",
       ">LC522974.1 Severe acute respiratory syndrome c...    0    0    0    0  \n",
       ">LC522975.1 Severe acute respiratory syndrome c...    0    0    0    0  "
      ]
     },
     "execution_count": 374,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 323,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pA</th>\n",
       "      <th>pT</th>\n",
       "      <th>pC</th>\n",
       "      <th>pG</th>\n",
       "      <th>pAA</th>\n",
       "      <th>pAT</th>\n",
       "      <th>pAC</th>\n",
       "      <th>pAG</th>\n",
       "      <th>pTA</th>\n",
       "      <th>pTA</th>\n",
       "      <th>pTC</th>\n",
       "      <th>pTG</th>\n",
       "      <th>pCA</th>\n",
       "      <th>pCT</th>\n",
       "      <th>pCC</th>\n",
       "      <th>pCG</th>\n",
       "      <th>pGA</th>\n",
       "      <th>pGT</th>\n",
       "      <th>pGC</th>\n",
       "      <th>pGG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>6.800000e+01</td>\n",
       "      <td>6.800000e+01</td>\n",
       "      <td>6.800000e+01</td>\n",
       "      <td>6.800000e+01</td>\n",
       "      <td>68.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>6.800000e+01</td>\n",
       "      <td>6.800000e+01</td>\n",
       "      <td>6.800000e+01</td>\n",
       "      <td>68.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>68.0</td>\n",
       "      <td>68.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>9.758109e-03</td>\n",
       "      <td>7.421660e-03</td>\n",
       "      <td>1.250687e-02</td>\n",
       "      <td>2.748763e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.264350e-04</td>\n",
       "      <td>2.036121e-05</td>\n",
       "      <td>1.501248e-03</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>1.048573e-17</td>\n",
       "      <td>5.242864e-18</td>\n",
       "      <td>1.048573e-17</td>\n",
       "      <td>1.092263e-19</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.310716e-18</td>\n",
       "      <td>2.389326e-20</td>\n",
       "      <td>8.738106e-19</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>9.758109e-03</td>\n",
       "      <td>7.421660e-03</td>\n",
       "      <td>1.250687e-02</td>\n",
       "      <td>2.748763e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.264350e-04</td>\n",
       "      <td>2.036121e-05</td>\n",
       "      <td>1.501248e-03</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>9.758109e-03</td>\n",
       "      <td>7.421660e-03</td>\n",
       "      <td>1.250687e-02</td>\n",
       "      <td>2.748763e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.264350e-04</td>\n",
       "      <td>2.036121e-05</td>\n",
       "      <td>1.501248e-03</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>9.758109e-03</td>\n",
       "      <td>7.421660e-03</td>\n",
       "      <td>1.250687e-02</td>\n",
       "      <td>2.748763e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.264350e-04</td>\n",
       "      <td>2.036121e-05</td>\n",
       "      <td>1.501248e-03</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>9.758109e-03</td>\n",
       "      <td>7.421660e-03</td>\n",
       "      <td>1.250687e-02</td>\n",
       "      <td>2.748763e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.264350e-04</td>\n",
       "      <td>2.036121e-05</td>\n",
       "      <td>1.501248e-03</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>9.758109e-03</td>\n",
       "      <td>7.421660e-03</td>\n",
       "      <td>1.250687e-02</td>\n",
       "      <td>2.748763e-04</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>9.264350e-04</td>\n",
       "      <td>2.036121e-05</td>\n",
       "      <td>1.501248e-03</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 pA            pT            pC            pG   pAA   pAT  \\\n",
       "count  6.800000e+01  6.800000e+01  6.800000e+01  6.800000e+01  68.0  68.0   \n",
       "mean   9.758109e-03  7.421660e-03  1.250687e-02  2.748763e-04   0.0   0.0   \n",
       "std    1.048573e-17  5.242864e-18  1.048573e-17  1.092263e-19   0.0   0.0   \n",
       "min    9.758109e-03  7.421660e-03  1.250687e-02  2.748763e-04   0.0   0.0   \n",
       "25%    9.758109e-03  7.421660e-03  1.250687e-02  2.748763e-04   0.0   0.0   \n",
       "50%    9.758109e-03  7.421660e-03  1.250687e-02  2.748763e-04   0.0   0.0   \n",
       "75%    9.758109e-03  7.421660e-03  1.250687e-02  2.748763e-04   0.0   0.0   \n",
       "max    9.758109e-03  7.421660e-03  1.250687e-02  2.748763e-04   0.0   0.0   \n",
       "\n",
       "        pAC   pAG   pTA   pTA           pTC           pTG           pCA   pCT  \\\n",
       "count  68.0  68.0  68.0  68.0  6.800000e+01  6.800000e+01  6.800000e+01  68.0   \n",
       "mean    0.0   0.0   0.0   0.0  9.264350e-04  2.036121e-05  1.501248e-03   0.0   \n",
       "std     0.0   0.0   0.0   0.0  1.310716e-18  2.389326e-20  8.738106e-19   0.0   \n",
       "min     0.0   0.0   0.0   0.0  9.264350e-04  2.036121e-05  1.501248e-03   0.0   \n",
       "25%     0.0   0.0   0.0   0.0  9.264350e-04  2.036121e-05  1.501248e-03   0.0   \n",
       "50%     0.0   0.0   0.0   0.0  9.264350e-04  2.036121e-05  1.501248e-03   0.0   \n",
       "75%     0.0   0.0   0.0   0.0  9.264350e-04  2.036121e-05  1.501248e-03   0.0   \n",
       "max     0.0   0.0   0.0   0.0  9.264350e-04  2.036121e-05  1.501248e-03   0.0   \n",
       "\n",
       "        pCC   pCG   pGA   pGT   pGC   pGG  \n",
       "count  68.0  68.0  68.0  68.0  68.0  68.0  \n",
       "mean    0.0   0.0   0.0   0.0   0.0   0.0  \n",
       "std     0.0   0.0   0.0   0.0   0.0   0.0  \n",
       "min     0.0   0.0   0.0   0.0   0.0   0.0  \n",
       "25%     0.0   0.0   0.0   0.0   0.0   0.0  \n",
       "50%     0.0   0.0   0.0   0.0   0.0   0.0  \n",
       "75%     0.0   0.0   0.0   0.0   0.0   0.0  \n",
       "max     0.0   0.0   0.0   0.0   0.0   0.0  "
      ]
     },
     "execution_count": 323,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "metadata": {},
   "outputs": [],
   "source": [
    "#standardization\n",
    "#Useful: https://stackoverflow.com/questions/40758562/can-anyone-explain-me-standardscaler\n",
    "scaler = StandardScaler()\n",
    "std_data = scaler.fit_transform(df.values)\n",
    "df_std = pd.DataFrame(std_data, index=df.index, columns=df.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 277,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>pA</th>\n",
       "      <th>pT</th>\n",
       "      <th>pC</th>\n",
       "      <th>pG</th>\n",
       "      <th>pAA</th>\n",
       "      <th>pAT</th>\n",
       "      <th>pAC</th>\n",
       "      <th>pAG</th>\n",
       "      <th>pTA</th>\n",
       "      <th>pTA</th>\n",
       "      <th>pTC</th>\n",
       "      <th>pTG</th>\n",
       "      <th>pCA</th>\n",
       "      <th>pCT</th>\n",
       "      <th>pCC</th>\n",
       "      <th>pCG</th>\n",
       "      <th>pGA</th>\n",
       "      <th>pGT</th>\n",
       "      <th>pGC</th>\n",
       "      <th>pGG</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>&gt;LC521925.1 Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/AI/I-004/2020 RNA, complete genome\\n</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;LC522972.1 Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/KY/V-029/2020 RNA, complete genome\\n</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;LC522973.1 Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-012/2020 RNA, complete genome\\n</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;LC522974.1 Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-501/2020 RNA, complete genome\\n</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;LC522975.1 Severe acute respiratory syndrome coronavirus 2 2019-nCoV/Japan/TY/WK-521/2020 RNA, complete genome\\n</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>-1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                     pA   pT   pC   pG  pAA  \\\n",
       ">LC521925.1 Severe acute respiratory syndrome c...  0.0  1.0 -1.0 -1.0  1.0   \n",
       ">LC522972.1 Severe acute respiratory syndrome c...  0.0  1.0 -1.0 -1.0  1.0   \n",
       ">LC522973.1 Severe acute respiratory syndrome c...  0.0  1.0 -1.0 -1.0  1.0   \n",
       ">LC522974.1 Severe acute respiratory syndrome c...  0.0  1.0 -1.0 -1.0  1.0   \n",
       ">LC522975.1 Severe acute respiratory syndrome c...  0.0  1.0 -1.0 -1.0  1.0   \n",
       "\n",
       "                                                    pAT  pAC  pAG  pTA  pTA  \\\n",
       ">LC521925.1 Severe acute respiratory syndrome c...  0.0  1.0  1.0 -1.0 -1.0   \n",
       ">LC522972.1 Severe acute respiratory syndrome c...  0.0  1.0  1.0 -1.0 -1.0   \n",
       ">LC522973.1 Severe acute respiratory syndrome c...  0.0  1.0  1.0 -1.0 -1.0   \n",
       ">LC522974.1 Severe acute respiratory syndrome c...  0.0  1.0  1.0 -1.0 -1.0   \n",
       ">LC522975.1 Severe acute respiratory syndrome c...  0.0  1.0  1.0 -1.0 -1.0   \n",
       "\n",
       "                                                    pTC  pTG  pCA  pCT  pCC  \\\n",
       ">LC521925.1 Severe acute respiratory syndrome c... -1.0  1.0  1.0  1.0 -1.0   \n",
       ">LC522972.1 Severe acute respiratory syndrome c... -1.0  1.0  1.0  1.0 -1.0   \n",
       ">LC522973.1 Severe acute respiratory syndrome c... -1.0  1.0  1.0  1.0 -1.0   \n",
       ">LC522974.1 Severe acute respiratory syndrome c... -1.0  1.0  1.0  1.0 -1.0   \n",
       ">LC522975.1 Severe acute respiratory syndrome c... -1.0  1.0  1.0  1.0 -1.0   \n",
       "\n",
       "                                                    pCG  pGA  pGT  pGC  pGG  \n",
       ">LC521925.1 Severe acute respiratory syndrome c...  1.0  1.0  1.0  0.0  1.0  \n",
       ">LC522972.1 Severe acute respiratory syndrome c...  1.0  1.0  1.0  0.0  1.0  \n",
       ">LC522973.1 Severe acute respiratory syndrome c...  1.0  1.0  1.0  0.0  1.0  \n",
       ">LC522974.1 Severe acute respiratory syndrome c...  1.0  1.0  1.0  0.0  1.0  \n",
       ">LC522975.1 Severe acute respiratory syndrome c...  1.0  1.0  1.0  0.0  1.0  "
      ]
     },
     "execution_count": 277,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_std.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
